airflow/models/variable.py

```diff
@@ -82,7 +83,7 @@ def val(cls):  # pylint: disable=no-self-argument
         return synonym('_val', descriptor=property(cls.get_val, cls.set_val))
 
     @classmethod
-    def setdefault(cls, key, default, deserialize_json=False):
+    def setdefault(cls, key, default, deserialize_json=False, deserialize_yaml=False):
```

Comment:
I wonder if we could do something better than these two vars.

Hmmmm.

How does  UI display one of these YAML serialized variables?

Reply:
I was thinking about that, if we want to support more than one format, would that make sense to have a parameter like serialize /deserialize of type str, which inputs could be one of json, yaml, etc.? Don't like this because it's not backward compatible. Or this could have been another argument and original would eventually be removed. 
I can attach some screenshots of UI tomorrow.